extensions/elytron-security/runtime/src/main/java/io/quarkus/elytron/security/runtime/ElytronRecorder.java

```diff
@@ -39,35 +30,20 @@ public void setDomainForIdentityProvider(BeanContainer bc, RuntimeValue<Security
      * @param realmName - the default realm name
      * @param realm - the default SecurityRealm
      * @return a runtime value for the SecurityDomain.Builder
-     * @throws Exception on any error
      */
-    public RuntimeValue<SecurityDomain.Builder> configureDomainBuilder(String realmName, RuntimeValue<SecurityRealm> realm)
-            throws Exception {
+    public RuntimeValue<SecurityDomain.Builder> configureDomainBuilder(String realmName, RuntimeValue<SecurityRealm> realm) {
         log.debugf("buildDomain, realm=%s", realm.getValue());
 
         SecurityDomain.Builder domain = SecurityDomain.builder()
 
                 .addRealm(realmName, realm.getValue())
 
-                .setRoleDecoder(new RoleDecoder() {
-                    @Override
-                    public Roles decodeRoles(AuthorizationIdentity authorizationIdentity) {
-                        return CDI.current().select(DefaultRoleDecoder.class).get().decodeRoles(authorizationIdentity);
-                    }
-                })
+                .setRoleDecoder(authorizationIdentity -> CDI.current()
+                        .select(DefaultRoleDecoder.class).get()
+                        .decodeRoles(authorizationIdentity))
```

Comment:
Please don't do that, we are not using lambdas in runtime code because it consumes memory.

Reply:
I didn't know. Will revert immediately